nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,147 @@
+                    mode = "OnRelease";
+                    action = {
+                      type = "Keypress";
+                      keys = [ "KEY_DOWN" ];
+                    };
+                  }
+                  {
+                    direction = "Left";
+                    mode = "OnRelease";
+                    action = {
+                      type = "CycleDPI";
+                      dpis = [ 400 600 800 1000 1200 1400 1600 ];
+                    };
+                  }
+                  {
+                    direction = "Right";
+                    mode = "OnRelease";
+                    action = { type = "ToggleSmartshift"; };
+                  }
+                  {
+                    direction = "None";
+                    mode = "NoPress";
+                  }
+                ];
+              };
+            }
+            {
+              cid = "0xc4";
+              action = {
+                type = "Keypress";
+                keys = [ "KEY_A" ];
+              };
+            }
+          ];
+        }];
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc."logid.cfg".source =
+      format.generate "logid.cfg" cfg.settings;
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      documentation = [ "https://github.com/PixlOne/logiops" ];
+
+      wantedBy = [ "multi-user.target" ];
+
+      startLimitIntervalSec = 0;
+      after = [ "multi-user.target" ];
+      wants = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/logid";
+        Restart = "always";
+        User = "root";
+
+        RuntimeDirectory = "logiops";
+
+        CapabilityBoundingSet = [ "CAP_SYS_NICE" ];
+        DeviceAllow = [ "/dev/uinput rw" "char-hidraw rw" ];
+        ProtectClock = true;
+        PrivateNetwork = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        PrivateUsers = true;
+        PrivateMounts = true;
+        PrivateTmp = true;
+        RestrictNamespaces = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        MemoryDenyWriteExecute = true;
+        RestrictRealtime = true;
+        LockPersonality = true;
+        ProtectProc = "invisible";
+        SystemCallFilter = [ "nice" "@system-service" "~@privileged" ];
+        RestrictAddressFamilies = [ "AF_NETLINK" "AF_UNIX" ];
+        RestrictSUIDSGID = true;
+        NoNewPrivileges = true;
+        ProtectSystem = "strict";
+        ProcSubset = "pid";
+        UMask = "0077";
+      };
+    };
+
+    # Add a `udev` rule to restart `logiops` when the mouse is connected
+    # https://github.com/PixlOne/logiops/issues/239#issuecomment-1044122412
+    services.udev.extraRules = ''
+      ACTION=="add", SUBSYSTEM=="input", ATTRS{id/vendor}=="046d", RUN{program}="${pkgs.systemd}/bin/systemctl --no-block try-restart logiops.service"
```

Comment:
```suggestion
      ACTION=="add", SUBSYSTEM=="input", ATTRS{id/vendor}=="046d", RUN{program}="${config.systemd.package}/bin/systemctl --no-block try-restart logiops.service"
```

Reply:
Is this so the service will (potentially) use an overridden version of `systemd`?